src/SonarQube.TeamBuild.PreProcessor/AnalysisConfigGenerator.cs

```diff
@@ -67,6 +67,13 @@ public static class AnalysisConfigGenerator
                 SonarQubeHostUrl = localSettings.SonarQubeUrl
             };
 
+            // Store whether or not a credential was passed in the command line. This is used to fail-fast when it is provided
+            // in the begin step but not the end step.
+            if (localSettings.CmdLineProperties.TryGetProperty(SonarProperties.SonarUserName, out var _))
+            {
+                config.SetConfigValue(ConfigSettingsExtensions.IsUsingCommandLineCredentialsKey, null);
```

Comment:
I'm not a fan of empty property bag entries i.e. where the only important thing is the existence or not of the key.
It makes it harder to tell if the data in the file is correct or not just by looking at the file - you have to know what the code that handles it is doing.
It's not major, but the data in the file would be more self-describing (and would make more sense to the Support team) if either it had a boolean value (i.e. extra code to parse the value safely), or if the data was held in a first-class property rather than the property bag.